When using go/install with cache:true, existing installation is not properly cleaned

Hello @felicianotech ! I've been trying to set-up CI for a project and I encountered the following issue

We are using a machine executor and then trying to install go through the command provided by the orb: 
```yaml
machine:
  image: ubuntu-2004:202101-01
steps:
    - checkout
    - go/install:
        cache: true
        version: 1.16.3
```

Without using the cache, the install script properly cleans the existing go installation (`1.15.6` if I remember well) before installing the new version.
But if we set `cache:true` it seems it is not getting cleaned properly as we get this kind of errors when using `go` after:

```
/usr/local/go/src/runtime/mgcsweepbuf.go:87:80: memstats.gc_sys undefined (type mstats has no field or method gc_sys)
/usr/local/go/src/runtime/mgcsweepbuf.go:106:102: memstats.gc_sys undefined (type mstats has no field or method gc_sys)
```

Which refers to a file removed in go 1.16, and is symptomatic of a badly cleaned installation.

Would it be possible to properly clean the existing installation when using the cached go binary ? It might be on purpose to avoid some other issues I am not aware of.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -44,6 +44,8 @@ workflows:
       - int-test-cimg-go
       - int-test-cimg-base
       - int-test-macos-executor
+      - int-test-vm-linux
+      - int-test-dirty-cache
 
       # publish a semver version of the orb. relies on
       # the commit subject containing the text "[semver:patch|minor|major|skip]"
@@ -63,6 +65,8 @@ workflows:
             - int-test-cimg-go
             - int-test-cimg-base
             - int-test-macos-executor
+            - int-test-vm-linux
+            - int-test-dirty-cache
           filters:
             branches:
               only: master
@@ -110,3 +114,37 @@ jobs:
       - run: go version && go build ./...
       - go/save-cache:
           key: "integration"
+  int-test-vm-linux:
+    machine:
+      image: ubuntu-2004:202104-01
+    steps:
+      - run:
+          name: "Check out sample project."
+          command: git clone https://github.com/CircleCI-Public/circleci-demo-go.git ~/project
+      - go/install:
+          version: 1.16.4
+      - go/load-cache:
+          key: "integration"
+      - go/mod-download
+      - run: go version && go build ./...
+      - go/save-cache:
+          key: "integration"
+  # test to make sure the dirty cache issue doesn't crop back up
+  int-test-dirty-cache:
+    machine:
+      image: ubuntu-2004:202104-01
+    steps:
+      - run:
+          name: "Check out sample project."
+          command: git clone https://github.com/CircleCI-Public/circleci-demo-go.git ~/project
+      - go/install:
+          version: 1.15.1
+      - go/load-cache:
+          key: "integration"
+      - go/mod-download
+      - run: go version && go build ./...
+      - go/save-cache:
+          key: "integration"
+      - go/install:
+          version: 1.16.4
+      - run: go version && go build ./...

--- a/src/commands/install.yml
+++ b/src/commands/install.yml
@@ -13,7 +13,7 @@ parameters:
     description: |
       String to use in cache key. Typically overriden when needed to bust cache.
     type: string
-    default: "v1"
+    default: "v2"
 steps:
   - os-detect/init
   - when:
@@ -23,6 +23,7 @@ steps:
         - run:
             name: "Prep cache restore"
             command: |
+              $SUDO rm -rf /usr/local/go
               $SUDO mkdir -p /usr/local/go
               $SUDO chown -R $(whoami): /usr/local/go
         - restore_cache:

